server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -715,7 +715,7 @@ public String getSecondaryStorageURL(SnapshotVO snapshot) {
     @Override
     public boolean deleteSnapshotDirsForAccount(long accountId) {
 
-        List<VolumeVO> volumes = _volsDao.findByAccount(accountId);
+        List<VolumeVO> volumes = _volsDao.findIncludingRemovedByAccount(accountId);
```

Comment:
Sorry my lack of knowledge here. Do snapshots have an entry in the `volumes` table?

Reply:
have to check, you might just have found an old hidden bug , @rafaelweingartner . On the other hand I have seen stranger things, a template is a volume of sorts isn't it.

Reply:
I just checked, and indeed snapshots have an entry in the `volumes` table. On the other hand, I could not find entries for volumes, but this does not affect the PR.